nat/traversal/pinger.go

```diff
@@ -161,58 +161,33 @@ func (p *Pinger) PingProvider(ip string, localPorts, remotePorts []int, proxyPor
 	return localPort, remotePort, err
 }
 
-func (p *Pinger) ping(conn *net.UDPConn, ttl int, stop <-chan struct{}) error {
-	// Windows detects that 1 TTL is too low and throws an exception during send
-	i := 0
-
+func (p *Pinger) ping(conn *net.UDPConn, remoteAddr *net.UDPAddr, ttl int, stop <-chan struct{}) error {
 	err := ipv4.NewConn(conn).SetTTL(ttl)
 	if err != nil {
 		return errors.Wrap(err, "pinger setting ttl failed")
 	}
 
-	for {
+	for i := 1; time.Duration(i)*p.pingConfig.Interval < p.pingConfig.Timeout; i++ {
```

Comment:
Shouldn't this be `i * Interval`? You're multiplying nanos with a Duration, I'm not sure what's the result :)

Reply:
Also, the whole condition is only approximate, because the loop code will also consume time. It would be better to simply use a timer, I guess.

Reply:
Changed it to be more readable. We don't even need `i` here anymore.